tests/kohana/ExceptionTest.php

```diff
@@ -96,4 +96,108 @@ public function test_text($exception, $expected)
 	{
 		$this->assertEquals($expected, Kohana_Exception::text($exception));
 	}
+
+	/**
+	 * Test if Kohana_Exception logs exceptions
+	 *
+	 * @test
+	 * @dataProvider provider_text
+	 * @param object $exception exception to test
+	 * @param string $expected  expected output
+	 */
+	public function test_logs_exception($exception, $expected)
+	{
+		$writer = new Kohana_ExceptionTest_Log_Writer_Memory();
+		Kohana::$log->attach($writer);
+		Kohana_Exception::log($exception);
+		$this->assertSame($expected, $writer->messages[0]['body']);
+
+		// clean up
+		Kohana::$log->detach($writer);
+	}
+
+	/**
+	 * Test if Kohana_Exception logs exceptions if Kohana::$log is not
+	 * a Kohana_Log
+	 *
+	 * @test
+	 * @dataProvider provider_text
+	 * @param object $exception exception to test
+	 * @param string $expected  expected output
+	 */
+	public function test_log_not_kohana_log($exception, $expected) {
+		$temp = Kohana::$log;
+		Kohana::$log = new Kohana_ExceptionTest_Log_Psr_Mock();
+		Kohana_Exception::log($exception);
+		$this->assertSame($expected, Kohana::$log->logs[0]['message']);
+
+		// clean up
+		Kohana::$log = $temp;
```

Comment:
@enov I would move this to the `setUp` and `tearDown` method - as you have it the cleanup won't run if the assertion fails which may cause issues elsewhere.